src/lib/es2022.sharedmemory.d.ts

```diff
@@ -0,0 +1,86 @@
+     */
+    and(typedArray: BigInt64Array | BigUint64Array, index: number, value: bigint): bigint;
+
+    /**
+     * Replaces the value at the given position in the array if the original value equals the given
+     * expected value, returning the original value. Until this atomic operation completes, any
+     * other read or write operation against the array will block.
+     */
+    compareExchange(typedArray: BigInt64Array | BigUint64Array, index: number, expectedValue: bigint, replacementValue: bigint): bigint;
+
+    /**
+     * Replaces the value at the given position in the array, returning the original value. Until
+     * this atomic operation completes, any other read or write operation against the array will
+     * block.
+     */
+    exchange(typedArray: BigInt64Array | BigUint64Array, index: number, value: bigint): bigint;
+
+    /**
+     * Returns the value at the given position in the array. Until this atomic operation completes,
+     * any other read or write operation against the array will block.
+     */
+    load(typedArray: BigInt64Array | BigUint64Array, index: number): bigint;
+
+    /**
+     * Stores the bitwise OR of a value with the value at the given position in the array,
+     * returning the original value. Until this atomic operation completes, any other read or write
+     * operation against the array will block.
+     */
+    or(typedArray: BigInt64Array | BigUint64Array, index: number, value: bigint): bigint;
+
+    /**
+     * Stores a value at the given position in the array, returning the new value. Until this
+     * atomic operation completes, any other read or write operation against the array will block.
+     */
+    store(typedArray: BigInt64Array | BigUint64Array, index: number, value: bigint): bigint;
+
+    /**
+     * Subtracts a value from the value at the given position in the array, returning the original
+     * value. Until this atomic operation completes, any other read or write operation against the
+     * array will block.
+     */
+    sub(typedArray: BigInt64Array | BigUint64Array, index: number, value: bigint): bigint;
+
+    /**
+     * If the value at the given position in the array is equal to the provided value, the current
+     * agent is put to sleep causing execution to suspend until the timeout expires (returning
+     * `"timed-out"`) or until the agent is awoken (returning `"ok"`); otherwise, returns
+     * `"not-equal"`.
+     */
+    wait(typedArray: BigInt64Array, index: number, value: bigint, timeout?: number): "ok" | "not-equal" | "timed-out";
+
+    /**
+     * A non-blocking, asynchronous version of wait and usable on the main thread.
+     * Waits asynchronously on a shared memory location and returns a Promise
+     * @see wait
+     */
+    waitAsync(typedArray: BigInt64Array, index: number, value: bigint, timeout?: number): { async: false, value: "ok" | "not-equal" | "timed-out" } | { async: true, value: Promise<"ok" | "not-equal" | "timed-out"> };

```

Comment:
1. `typedArray` can be `Int32Array` too, right? At least according to MDN.
2. waitAsync is only available on Chrome-based browsers, not Firefox or Safari. Even though it's stage 3, I'm not super happy about adding it since for browser features outside the standard we require at least two implementations.

Reply:
Yeah you are right! Should I adjust for other functions also ? like ``wait`` method also expects ``Int32Array``  or  ``BigInt64Array `` I don't know what would be better.
Only changing this one or all other methods...

Reply:
This PR/ Issue comes from an Issue from a lib that uses Deno. In deno there are no type def. for this method, thought that I could help that issue by adjusting this.
We should wait for Firefox or Safari to implement this in order to proceed with this pull request correct ? I think this makes sense :)